### ISSUE TITLE
Magento 2.2.0-rc22 static content deploy not compiling locale for child theme

### ISSUE THREAD

**Issue by ennostuurman** [magento/magento2#10766]
Child theme locale is not compiled to pub/static/frontend/themedir/locale after running bin/magento setup:static-content:deploy in Magento 2.2.0-rc22

### Preconditions
1. Magento 2.2.0-rc22 (Communit Edition) installed w Composer
2. nginx/1.10.0
3. PHP 7.0.15-0ubuntu0.16.04.4
4. MAGE_MODE = production
5. Theme and storeview setup as follows
  a. Create a custom parent theme
  b. Apply parent theme to default storeview (w/ locale en_US)
  c. Create a child theme
  d. Create new storeview 
  e. Configure different locale (e.g. nl_NL) to new storeview
  f. Apply child theme to new storeview

### Steps to reproduce
1. Run bin/magento setup:static-content:deploy in different variants (see attached screenshots).

### Expected result
1. Child theme locale should be compiled to pub/static 

### Actual result
1. Child theme locale is not compiled to pub/static

![magento2 2 0-rc22-theme-setup](https://user-images.githubusercontent.com/1906257/30029980-b20416c8-918b-11e7-9a79-0f9f19603c50.png)
![magento2 2 0-rc22-static-content-deploy-quick](https://user-images.githubusercontent.com/1906257/30029981-b206b1a8-918b-11e7-8843-7ddefac634c7.png)
![magento2 2 0-rc22-static-content-deploy-quick-theme-locale](https://user-images.githubusercontent.com/1906257/30029983-b20f8f1c-918b-11e7-9a97-ec2f548aa134.png)
![magento2 2 0-rc22-static-content-deploy-standard-theme-locale](https://user-images.githubusercontent.com/1906257/30029982-b20ef156-918b-11e7-8cd2-889a2900d717.png)
![magento2 2 0-rc22-static-content-deploy-compact-theme-locale](https://user-images.githubusercontent.com/1906257/30029984-b2145c04-918b-11e7-84b7-acc1202a51d8.png)

**Comment by TomashKhamlai** [comment]
Thank you for your submission. Didn't try to dive deep into this issue, but just noticed that you are using `root` user to perform CLI commands, also you are not trying to use
`php bin/magento setup:static-content:deploy`
instead of 
`bin/magento setup:static-content:deploy`
This [guide](http://devdocs.magento.com/guides/v2.2/install-gde/prereq/file-sys-perms-over.html) may be useful.

Is your issue in some way connected with default 'Luma' or 'Blank' theme?

**Comment by alankent** [comment]
Could you please provide a bit more information? Have a look in config.php and env.php and report what is in the "scopes" section. http://devdocs.magento.com/guides/v2.2/config-guide/prod/config-reference-configphp.html.  There were changes in 2.2 around this area. The official list of scopes is now in these files for pushing say from dev to prod. That might be related to your problem.

**Comment by ennostuurman** [comment]
@TomashKhamlai I haven't had file permission issues with bin/magento setup:static-content:deploy in this Docker environment before. So I feel it can't be the 'root' cause in this issue ;-) . 

I am giving you some extra info so we can figure out if this is a special case or a real M2 issue.

When I add an i18n folder to the web folder in the frontend-theme-en then Magento will compile the Dutch locale for both themes. See screenshot. So we need to figure out why in M2.2 a locale within a theme does trigger locale compilation for all themes and why a seperate child theme does not.

I suggest you first try yourself with a separate theme, if you have no issues with compiling at your end, then it's an issue within my setup or M2.2 deploy workflow.

Screenshot of adjusted theme with i18n folder:
![magento2 2 0-rc22-static-content-deploy-with-separate-i18n-folder-theme-structure](https://user-images.githubusercontent.com/1906257/30055131-bc32580a-922e-11e7-8283-ce755c21c16a.png)


Compilation screenshot with and without separate i18n folder in en theme:
![magento2 2 0-rc22-static-content-deploy-with-separate-i18n-folder](https://user-images.githubusercontent.com/1906257/30054644-02f33338-922d-11e7-81ee-dd1ef4a42ce4.png)

**Comment by jhruehl** [comment]
@ennostuurman Your shabby workaround solution with adding the logo to web/i18n/[area code]/images/ folder actually works. And though you still cannot specify the area and language of the deployment, you can exclude them with the --exclude-area and --exclude-language option.

But this definitely needs to be fixed before the release von Magento 2.2.

**Comment by jhruehl** [comment]
@ennostuurman So I deleted my last comment, because it lead to nowhere.

There is a much better tempory solution than adding logos to some specific folders, which just induces all languages set in that way to be compiled is to specify the theme you want to deploy with the wanted language. It's not perfect, but a working solution, which needs no adjustments.

Just run
'bin/magento setup:static-content:deploy theme=Custom/nl nl_NL'.
And remove your workaround solution beforehand. Otherwise you probably won't see a difference.

So you can still specify the compilation process in that way, but of course most of the other options for the deployment process don't really work the way they should.

**Comment by atsareva** [comment]
Hi all,
I faced with the same issue after updated magento with 2.2.0 version. 
I've created folder with logo image in the custom theme and seems like static deploy works fine with all specified languages. 
Thanks for @ennostuurman

**Comment by magento-engcom-team** [comment]
@ennostuurman , thanks for the report. Unfortunately, we can not reproduce this issue with specified steps.
Steps we did:
Magento 2.2.2
1. Create custom parent theme(based on blank)
2. Create child theme(basically empty, has only registration.php and theme.xml)
3. Create additional store view.
4. Assign parent theme to default store view(en_US)
5. Assign child theme to custom store view(nl_NL)
6. Enable production mode (bin/magento deploy:mode:set production)
Both themes have static files in en_US and nl_NL locale.
Please correct our steps, in case they are wrong, or try to reproduce this issue on latest 2.2.x(currently 2.2.2)

**Comment by ennostuurman** [comment]
@magento-engcom-team thx for your investigation, the use case that you tested (bin/magento deploy:mode:set production) does indeed produce the right output. But that is not the issue that I described.

Did you also test the different deploy strategies, http://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-static-view.html 'stand alone' in production mode? 

1. remove pub/static 
2. run a quick, standard and compact static:content:deploy ?

**Comment by magento-engcom-team** [comment]
@ennostuurman, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by ennostuurman** [comment]
@magento-engcom-team thanx for the feedback. It might be obvious but will you also test this with the 2.2 deployment pipeline http://devdocs.magento.com/guides/v2.2/config-guide/deployment/pipeline/ (i.e. config export including multi store views and locales) ?

**Comment by tux-rampage** [comment]
The languages must no longer be specified via the `-l` or `--language` flags, but as optional argument list.

This was changed and the input option was not removed causing an incorrect cli usage for `--help`. :facepalm: 

Use it like this, and the locales will be deployed:

```bash
./bin/magento setup:static-content:deploy en_US de_DE nl_NL
```

Never the less they should also be generated for `'all'` (i.e. when Locales are omitted)

**Comment by ennostuurman** [comment]
hey @tux-rampage thank you for chiming in. I didn't dig into this further for some time, this refers back to a release candidate version some while ago. I indeed didn't use the language flag anymore for quite some time. 

I didn't dig into the code but maybe @magento-engcom-team can shed some light on the supposed removal of the locale flag. Because if it has been removed than both the  `bin/magento se:st:de --help ` and dev docs documentation should be updated as well https://devdocs.magento.com/guides/v2.3/config-guide/cli/config-cli-subcommands-static-view.html . 

And hopefully we can then close this 2,5 yrs old issue.
